data_pipeline/twitter_ads_api/twitter_ads_api_pipeline.py

```diff
@@ -85,36 +84,87 @@ def get_param_dict_from_api_query_parameters(
     return result
 
 
+def get_final_end_date(
+    api_query_parameters_config: TwitterAdsApiApiQueryParametersConfig,
+    initial_start_date_value: datetime
+) -> datetime:
+    yesterday_date = get_yesterdays_date()
+    end_date_by_period = initial_start_date_value + timedelta(
+        days=api_query_parameters_config.parameter_values.ending_period_per_day
+    )
+    if end_date_by_period > yesterday_date:
+        final_end_date_value = yesterday_date
+    else:
+        final_end_date_value = end_date_by_period
+    return final_end_date_value
+
+
+def get_end_date_value_of_period(
+    start_date_value: datetime,
+    final_end_date_value: datetime,
+    days_in_period: int
+) -> datetime:
+    period_end_date_value = start_date_value + timedelta(days=days_in_period)
+    if period_end_date_value > final_end_date_value:
```

Comment:
maybe this if / else block could be reused somehow. If those were numbers, then you could use `min`. So maybe `get_min_date(date1: date, date2: date) -> date`?

Or including adding the days:

`add_days_with_max_date(start_date: date, days: int, max_date: date) -> date`?